arch/arm/src/common/arm_internal.h

```diff
@@ -306,14 +307,10 @@ void arm_boot(void);
 
 /* Context switching */
 
-void arm_copyfullstate(uint32_t *dest, uint32_t *src);
-#ifdef CONFIG_ARCH_FPU
-void arm_copyarmstate(uint32_t *dest, uint32_t *src);
-#endif
 uint32_t *arm_decodeirq(uint32_t *regs);
 int  arm_saveusercontext(uint32_t *saveregs);
 void arm_fullcontextrestore(uint32_t *restoreregs) noreturn_function;
-void arm_switchcontext(uint32_t *saveregs, uint32_t *restoreregs);
+void arm_switchcontext(uint32_t **saveregs, uint32_t *restoreregs);
```

Comment:
Do we need to add `FAR` to all the pointers?

Reply:
Yes, I will send a independent patch to complete FAR

Reply:
Should we add `FAR` to arch specific code ? It's no sense on arm (empty defination).

Reply:
It's better to remove FAR just like what we have done for riscv